op-node/rollup/derive/batch.go

```diff
@@ -39,12 +39,20 @@ type Batch interface {
 	LogContext(log.Logger) log.Logger
 }
 
-// BatchData is a composition type that contains raw data of each batch version.
-// It has encoding & decoding methods to implement typed encoding.
+// BatchData is used to represent the typed encoding & decoding.
+// and wraps arount a single interface InnerBatchData.
 type BatchData struct {
-	BatchType int
-	SingularBatch
-	RawSpanBatch
+	inner InnerBatchData
+}
+
+// InnerBatchData is the underlying data of a BatchData.
+// This is implemented by SingularBatch and RawSpanBatch.
```

Comment:
(optional) It would be nice to have a more detailed explanation of the difference between `Batch` and `InnerBatchData` and the design intent.